### ISSUE TITLE
Support OpenBao

### ISSUE THREAD

**Issue by QuanTran91** [external-secrets/external-secrets#3664]
**Is your feature request related to a problem? Please describe.**
AFAIK, the OpenBao is a fork of Vault. I'm not sure if external-secret supports it.

**Describe the solution you'd like**
If it supports OpenBao, that would be great if you guys adding it in your document

**Comment by QuanTran91** [comment]
Yes, I think so. But it would be great if external-secrets mention it in their document. And also support it in their roadmap

**Comment by gusfcarvalho** [comment]
> But it would be great if external-secrets mention it in their document. And also support it in their roadmap

IMO, it only makes sense to mention in the document if we dedicate proper support for it (client libraries, logic, etc). So far, no one from the community aside from you really asked that 😅  OpenBAO support with ESO is a feature from OpenBAO, and not really an ESO feature. 

Hence, I'm not sure we should 'add' it as a feature. Maybe as a guide "connecting to openBAO"

**Comment by eyenx** [comment]
I just tested externalSecretsOperator using the Vault Provider with OpenBao 2.2.0 and it just works. 

I'm willing to contribute to the documentation to mention OpenBao or even support adding a dedicated Provider to ESO as Bao & Vault code base will derive from each other sooner or later.

**Comment by snuxoll** [comment]
One thing that would be specifically beneficial for ESO is the support for [inline authentication](https://github.com/openbao/openbao/issues/1301) that landed in OpenBao 2.4. My own deployments aren't anywhere needing this kind of scalability, but just like OpenTofu has already diverged from Terraform there's no sign that OpenBao will be slowing down their own development of features that (while compatible) work best with client support.

**Comment by evrardj-roche** [comment]
This issue has been marked as Stale and will be closed soon if there is no activity. Please update the issue or engage with it to keep it open. It is fine to re-open this issue in the future if you want to work on this.

**Comment by eyenx** [comment]
I'm suggesting to close this. OpenBao is supported over the Vault provider and is also documented now. We will open a new issue to build our own provider looking also at inline Authentication and supporting other secrets engines.

**Comment by Skarlso** [comment]
Sounds good to me. :) ❤️

**Comment by evrardj-roche** [comment]
I had a conversation with some adfinis folks at fosdem, to make openBao a first citizen. This means a few things:
- We can implement features that are specific to openbao 
- A more friendly licensing
- More test cases (namespaces!)
- We could be the official operator for openbao, instead of the bao-secret-operator.

I think this topic can be closed (because we _do_ support openbao), but we should keep in mind that making it a first citizen might be a benefit for us in the long term. I will try to work with the adfinis folks to improve our integrations and scope (I am using ESO with openbao myself after all!)

**Comment by phil9909** [comment]
I can start working on this

**Comment by evrardj-roche** [comment]
@phil9909 if you need any help, I can be there for you.

**Comment by phil9909** [comment]
@evrardj-roche Thanks, I'm making good progress.

I'll have to update the docs and will open a PR after that.

**Comment by gusfcarvalho** [comment]
> A more friendly licensing

Re: licensinng, this really is only an objective if people are shipping eso deployments without vault codebase. Which is a weak argument as our upstream ESO distribution is deployed with every single provider by default.

**Comment by evrardj-roche** [comment]
> > A more friendly licensing
> 
> Re: licensinng, this really is only an objective if people are shipping eso deployments without vault codebase. Which is a weak argument as our upstream ESO distribution is deployed with every single provider by default.

so far... :) 
But I agree it's a weak point. 

It was just because I had to look at CNCF exceptions to see if we are still fine, which would not be required here.

We could even think of removing hashicorp module is bao modules are both compatible openbao and hashicorp vault... But that's definitely getting ahead of ourselves there.

**Comment by QuanTran91** [comment]
How is it going with the openbao support? Thanks

**Comment by Skarlso** [comment]
We are going to do a bit of sequential approach because the initial PR was just way too big. :) But it's ongoing and will definitely happen :)
